sklearn/decomposition/tests/test_sparse_pca.py

```diff
@@ -237,5 +237,6 @@ def test_pca_vs_spca():
 def test_spca_deprecation_warning(spca):
     rng = np.random.RandomState(0)
     Y, _, _ = generate_toy_data(3, 10, (8, 8), random_state=rng)
-    with pytest.warns(DeprecationWarning, match="normalize_components"):
-        spca(normalize_components=False).fit(Y)
+    warn_message ="normalize_components"
```

Comment:
why not X ? Y is a bit confusing

Reply:
Indeed but all the other tests are with Y (and equally confusing) so I kept the local convention. Should I change this ?